R/arg_reconcile.R

```diff
@@ -0,0 +1,117 @@
+#'                \code{utils::read.table})
+#' @param ...     An arbitrary list of named arguments (unnamed ones will be 
+#'                ignored). Arguments in \code{.args} are overridden by
+#'                arguments of the same name (if any) in \code{...}
+#' @param .args   A list or \code{alist} of named arguments, to be merged 
+#'                with \code{...}. Arguments in \code{.args} are overridden by
+#'                arguments of the same name (if any) in \code{...}
+#' @param .docall If set to \code{TRUE} will not only clean up the arguments but 
+#'                also execute \code{fn} with those arguments (\code{FALSE} by
+#'                default) and return the results
+#' @param .wtlist Whitelist. If not empty, only arguments named here will be 
+#'                permitted, and only if they satisfy the conditions implied by
+#'                the other arguments. Evaluated before \code{.remap}.
+#' @param .bklist Blacklist. If not empty, arguments named here will be removed 
+#'                even if they satisfy the conditions implied by the other 
+#'                arguments. Evaluated before \code{.remap}.
+#' @param .remap  An optional named character vector or named list of character 
+#'                values for standardizing arguments that play the same role but
+#'                have different names in different functions. Evaluated after
+#'                \code{.bklist} and \code{.wtlist}.
+#' @param .warn   Whether to issue a warning message (default) when invalid 
+#'                arguments need to be discarded.
+#' @param .error  If specified, should be the object to return in the event of
+#'                error. This object will have the error as its 
+#'                \code{error} attribute. If not specified an ordinary error is
+#'                thrown with an added hint on the documentation to read for 
+#'                troubleshooting. Ignored if \code{.docall} is \code{FALSE}.
+#'                The point of doing this is fault-tolerance-- if this function
+#'                is part of a lengthy process where you want to document an 
+#'                error but keep going, you can set \code{.error} to some object
+#'                of a compatible type. That object will be returned in the 
+#'                event of error and will have as its \code{"error"} attribute 
+#'                the error object.
+#' @param .finish A function to run on the result before returning it. Ignored 
+#'                if \code{.docall} is \code{FALSE}.
+#'
+#' @return Either a named list or the result of calling \code{fn} with the 
+#'         supplied arguments
+#'
+#' @examples
+#' 
+#' arg_reconcile(p.adjust, bla = "aaa", baz = "xzzz", n = 40, 
+#'                     q = c(.1, .2, .02, .3, 1, 0, 1), .remap = c(q = "p"))
+#' arg_reconcile(import, file = "hello.csv", file = "goodbye.csv", bla = "aaa", 
+#'               baz = "xzzz", format = "csv")
+#' 
+#' @export
```

Comment:
Main reaction at this point is that I don't think we need to export this function.

Reply:
Done!